addons/survey/controllers/main.py

```diff
@@ -460,7 +460,8 @@ def survey_get_question_image(self, survey_token, answer_token, question_id, sug
 
     @http.route('/survey/begin/<string:survey_token>/<string:answer_token>', type='json', auth='public', website=True)
     def survey_begin(self, survey_token, answer_token, **post):
-        """ Route used to start the survey user input and display the first survey page. """
+        """ Route used to start the survey user input and display the first survey page.
+        Returns an empty dict for the correct answers and the first page html. """
         access_data = self._get_access_data(survey_token, answer_token, ensure_token=True)
         if access_data['validity_code'] is not True:
             return {'error': access_data['validity_code']}
```

Comment:
Could you check that this works? It could work fine, as the response will be caught at another level than our handler; but then, we should do the same in `survey_submit`, right?

Reply:
But does the error work?